todo-app/src/TodoApp.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import TodoList from './TodoList';
+import TodoForm from './todoForm';
+import Moment from 'moment';
+import {loadTodos, saveTodos} from './storage';
+
+export default class TodoApp extends React.Component {
+
+    state = {
+       todos: loadTodos(),
+    };
+ 
+    setTodos(todos) {
+        this.setState({todos});
+        saveTodos(todos);
+    };
+
+    handleDoneToggle = (todoId) => {
+        
+        const newState = this.state.todos.map(
+            todo => {
+              if (todo.id === todoId) {
+                return {...todo, done: !todo.done};
+              } else {
+                return todo
+              }
+            }
+          );
+          
+        this.setTodos(newState);
+    };
+
+    newTodoWithText(text, deadline) {
+        const ids = this.state.todos.map(todo => todo.id)
+
+        return {
+            // id: Math.max(...ids) + 1, solving todos with same id 
+            id: ids.length === 0 ? 1 : Math.max(...ids) + 1,
+            task: text,
+            addedTime: Moment().format('YY-MM-DD'),
+            deadlineTime: deadline,
+            done: false,
+        }
+    };
+    
+    handleAddTodoClick = (todoText, deadline) => {
+
+        const todo = this.newTodoWithText(todoText, deadline);
+        this.setTodos([todo, ...this.state.todos]);
+        console.log("Adding todo :", todoText, "deadline is ", deadline);
+    };
+
+    //saves text after editing
+
+    handleSaveTodo = (id, task) => {
+
+        this.setTodos(this.state.todos.map(todo => {
+            if (todo.id === id) {
+              return {...todo, task};
+            } else {
+              return todo
+            }
+      }));
+    };
+
+    //saves deadline after editing
+    //both handleSave and handleSaveDeadline are not working at  
+    //sametime thats why i've used setTimeOut until updating to mobx
```

Comment:
Typically if you have to resort to such measures, you have some other problem in your code. In this case, why not have one single `handleSave` method, which can update multiple properties at once?

Reply:
The reason why it doesn't work in two goes, is because `setState` updates the state in the next cycle (that is, until JavaScript has finished executing all code). You should not call `setState` and then later use `this.state....` because you will still use the old version. 

But, all of that is not a problem if you would use one function to update multiple properties.